fastify.js

```diff
@@ -424,7 +424,11 @@ function build (options) {
 // Everything that need to be encapsulated must be handled in this function.
 function override (old, fn, opts) {
   const shouldSkipOverride = pluginUtils.registerPlugin.call(old, fn)
+
   if (shouldSkipOverride) {
+    // after every plugin registration we will overwrite the name
+    // that's why "current" has the correct meaning
+    old.currentPluginName = pluginUtils.getDisplayName(fn)
```

Comment:
I think this is the most problematic bit. Essentially we are overriding the plugin name every single time, losing the previous information. I think this would cause issues to our users, and I would prefer to not do it all, or add the ones where we skip override at the end, something like `myPlugin (fastify-cors, fastify-leveldb, fastify-auth)` instead.

Reply:
Fastify plugins don't use encapsulation. Where should we store that information? I don't understand your example at the end.

Reply:
Essentially, I do not want to lose the information that multiple non-encapsulated plugins can override the name.

How about:

```js
old.currentPluginName += pluginUtils.getDisplayName(fn)
```

Reply:
For me it's feels odd that a plugin can't have its scope when using plugins :smile:

> How about:
old.currentPluginName += pluginUtils.getDisplayName(fn)

It's better but this doesn't avoid the fact that the plugin name is overwritten and only correct (from consumer perspective) when the user access it synchronously.

Reply:
true. However it would represent the current list of plugins loaded at that encapsulation level, which will be representative anyway.

Reply:
I will update it accordingly.